ARCHITECTURE.md

```diff
@@ -51,34 +51,40 @@ https://github.com/replicatedhq/kurl-api
 
 The browser UI for creating and viewing kURL installers.
 
-#### Go API Proxy
+#### Typescript API
 
-This API proxies most requests directly to the Typescript API.
-Additionally, this API interacts with the object store and is responsible for assembling and streaming air-gap bundles back to the end-user.
+The API for creating kURL URLs and rendering installers.
+The API accepts a kURL Installer spec and returns a URL including a deterministic hash of the spec for installing a cluster based on that spec, for example https://k8s.kurl.sh/5e61e80.
+The API interacts stores these hashes in its relational database for retrieval.
+Additionally, the API is responsible for resolving the add-on version ("latest" and ".x") and rendering the spec when a script is requested, for example https://k8s.kurl.sh/5e61e80/install.sh or shorthand https://k8s.kurl.sh/5e61e80.
+The API reads the current kURL version from the object storage bucket (https://kurl-sh.s3.amazonaws.com/dist/VERSION) and uses that version to lookup add-on version information (e.g. https://kurl-sh.s3.amazonaws.com/dist/v2022.04.19-0/supported-versions-gen.json) for spec resolution.
 
-#### Typescript API
+#### Go API Proxy
 
-The API for creating and consuming installers. Interacts with the relational database.
+This API proxies most requests directly to the Typescript API.
+Additionally, this API is responsible for assembling and streaming air-gap bundles back to the end-user.
+Air-gap packages are assembled from individual add-on package archives (e.g. https://kurl-sh.s3.amazonaws.com/dist/v2022.04.19-0/rook-1.0.4.tar.gz) and streamed as a single archive back to the end user when requested (e.g. https://k8s.kurl.sh/bundle/5e61e80.tar.gz).
 
 #### Relational Database
 
 Stores installer hashes.
 
 #### Object Store
 
-Stores built kURL assets including add-on bundles.
+Stores built kURL assets including add-on package archives.
```

Comment:
It still not clear for me. Why? 
Ok. I build those to do the tests. However, it would be very nice to know how they are stored?
What are the formats? Could we add an example ?